### PR TITLE
[crd-generator] Fix fallback value of `Default` annotation in presence of multiple accessors

### LINKED ISSUE #5501: Default annotation not working as expected
### Describe the bug

I am using [the new `Default` annotation](https://github.com/fabric8io/kubernetes-client/issues/5410) to generate my CRDs with defaults. My spec class looks like this:
```java
@Data
@Jacksonized
@Builder
public class MyAppSpec {
        @Required
        private String name;
        @Default("1")
        private int replicaCount;
        @Required
        private String image;
        @Required
        private String imageTag;
        @Default("Always")
        private String imagePullPolicy;
        @Default("false")
        private boolean deployment;
}
```

And the generated CRD is missing the default values. The output looks like this:
```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: myapps.com.organisation.tree
spec:
  group: com.organisation.tree
  names:
    kind: MyApp
    plural: myapps
    singular: myapp
  scope: Namespaced
  versions:
  - name: v1
    schema:
      openAPIV3Schema:
        properties:
          spec:
            properties:
              deployment:
                type: boolean
              image:
                type: string
              imagePullPolicy:
                type: string
              imageTag:
                type: string
              name:
                type: string
              replicaCount:
                type: integer
            required:
            - name
            - image
            - imageTag
            type: object
          status:
            properties:
              ready:
                type: boolean
            type: object
        type: object
    served: true
    storage: true
    subresources:
      status: {}
```

### Fabric8 Kubernetes Client version

6.9.0

### Steps to reproduce

1. Create a spec POJO
2. Annotate fields with `@Default`
3. Generate the CRD

### Expected behavior

To have `default` field in the generated CRD

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.25.3@latest

### Environment

macOS

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andreaTP**
## Description
Fix #5501 

## Type of change
 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java [hunk 1]
```diff
@@ -558,7 +558,7 @@ public Property process() {
             LOGGER.debug("Description for property {} has already been contributed by: {}", name, descriptionContributedBy);
           }
         }
-        defaultValue = p.getDefault().orElse(null);
+        defaultValue = p.getDefault().orElse(defaultValue);
```
Comment (andreaTP) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:561]: for the records, this is the bug.

File: crd-generator/test/src/test/java/io/fabric8/crd/generator/zookeeper/v1/ZookeeperSpec.java [hunk 2]
```diff
@@ -15,14 +15,18 @@
...
 import io.fabric8.kubernetes.model.annotation.SpecReplicas;
 
+@Data
 public class ZookeeperSpec {
 
   @SpecReplicas
   private int size;
   @Required
   private String version;
+  @Default("false")
```
Comment (andreaTP) [inline crd-generator/test/src/test/java/io/fabric8/crd/generator/zookeeper/v1/ZookeeperSpec.java:30]: Please, take note that the issue was reproducible only here.

File: crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java [hunk 3]
```diff
@@ -154,6 +154,13 @@ void shouldAugmentPropertiesSchemaFromAnnotations() throws JsonProcessingExcepti
     assertNull(defaultValue.getMaximum());
     assertNull(defaultValue.getPattern());
 
+    final JSONSchemaProps defaultValue2 = spec.get("defaultValue2");
+    assertEquals("my-value2", YAML_MAPPER.writeValueAsString(defaultValue2.getDefault()).trim());
+    assertNull(defaultValue.getNullable());
+    assertNull(defaultValue.getMinimum());
+    assertNull(defaultValue.getMaximum());
+    assertNull(defaultValue.getPattern());
```
Comment (rohanKanojia) [inline crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java:162]: I might be wrong here but shouldn't you see asserting fields for `defaultValue2` here?

**Review by rohanKanojia — APPROVED** [review]

File: crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java (hunk 3 above)
Comment (andreaTP) [reply crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java:162]: good catch! fixed!

**Review by manusa — APPROVED** [review]

**Pull request merged**